test/e2e/framework/pod/wait.go

```diff
@@ -170,33 +171,30 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods, allowedN
 		// Clear API error from the last attempt in case the following calls succeed.
 		lastAPIError = nil
 
-		rcList, err := c.CoreV1().ReplicationControllers(ns).List(context.TODO(), metav1.ListOptions{})
+		rcList, err := c.CoreV1().ReplicationControllers(ns).List(context.Background(), metav1.ListOptions{})
```

Comment:
Side note: in https://github.com/onsi/ginkgo/issues/969 I am discussing how overall suite timeouts could be handled better. Right now, the suite gets killed and cleanup code doesn't run, which can, for example, lead to leaked volumes in the cloud provider.

The solution that I have in mind is that Ginkgo provides a context and tests must return ASAP once that context is done. It will take more time (and convincing of the upstream ginkgo author) before that is feasible, but one good first step would be to make all of our poll functions take a context. So instead of replacing `context.TODO()` with `context.Background()` here, keep the `context.TODO()` because `context.Background()` is not the end goal.